src/components/data-exploration/CellInfo.jsx

```diff
@@ -1,22 +1,37 @@
 import React from 'react';
-import _ from 'lodash';
 import { Card } from 'antd';
 import PropTypes from 'prop-types';
 
+const EM = 16; // px
+const CELLINFO_WIDTH = 200; // px
+const CELLINFO_Y_PADDING = 6 * EM;
+const Y_PADDING = 2 * EM;
+
+const cellInfoStyle = { fontSize: '0.75rem' };
+
 const CellInfo = (props) => {
-  const { coordinates, cellInfo } = props;
+  const {
+    containerWidth, containerHeight, coordinates, cellInfo,
+  } = props;
+
+  const numTextRows = (cellInfo.cellSets?.length || 0) + 2;
```

Comment:
similarly to the width, is there another way to know the height, or know when to invert without knowing the height?

Reply:
We can't know when to invert if we don't know whether the box will overflow outside the panel or not, and to know that, we need to know the position of the cursor and the height of the box. So there's no way of knowing when to invert if we don't know the height that the box is going to be.

Just like the width, I used an approximation for the height. The height of a line of text is usually 1em. So if we know the number of lines, then we'll know the height, assuming there is no text-overflow to the next line. This is what I used in these popups.